test/suite/test_checkpoint23.py

```diff
@@ -0,0 +1,280 @@
+        session = self.session
+        cursor = session.open_cursor(ds.uri)
+        for i in range(1, nrows + 1):
+            session.begin_transaction()
+            cursor[ds.key(i)] = secondary_value if i % 10 == 1 else primary_value
+            session.commit_transaction('commit_timestamp=' + self.timestamp_str(commit_ts))
+        cursor.close()
+
+    def large_removes(self, ds, nrows, commit_ts):
+        session = self.session
+        cursor = session.open_cursor(ds.uri)
+        for i in range(1, nrows + 1):
+            session.begin_transaction()
+            cursor.set_key(ds.key(i))
+            self.assertEqual(cursor.remove(), 0)
+            session.commit_transaction('commit_timestamp=' + self.timestamp_str(commit_ts))
+        cursor.close()
+
+    def test_checkpoint23(self):
+        uri = 'table:checkpoint23'
+        nrows = 10000
+
+        # Create a table.
+        ds = SimpleDataSet(self, uri, nrows,
+            key_format=self.key_format, value_format=self.value_format)
+        ds.populate()
+
+        # Pin oldest and stable timestamp to 1.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(1) +
+            ',stable_timestamp=' + self.timestamp_str(1))
+
+        # Use a seconary value large enough to be an overflow item.
+        value_a = "aaaaa" * 100
+        value_b = "bbbbb" * 10000
+
+        # Write a bunch of data at time 10.
+        self.large_updates(ds, nrows, value_a, value_b, 10)
+
+        # Delete it all at time 20.
+        self.large_removes(ds, nrows, 20)
+
+        # Mark it stable at 30 and checkpoint it.
+        self.conn.set_timestamp('stable_timestamp=' + self.timestamp_str(30))
+        self.session.checkpoint()
+
+        # Shut down and restart so nothing's in memory.
+        self.reopen_conn()
+
+        # Check how many overflow items we have.
+        initial_overflow_items = self.get_tree_stat(ds, stat.dsrc.btree_overflow)
+
+        # Shut down and restart again so nothing's in memory.
+        self.reopen_conn()
+
+        # Move oldest up so all the data is obsolete.
+        self.conn.set_timestamp('oldest_timestamp=' + self.timestamp_str(30))
+
+        # Do a checkpoint. Name the checkpoint to force things through.
+        self.session.checkpoint('name=pointy')
+
+        # Do another checkpoint to make sure the updated pages get reconciled.
+        # Otherwise the overflow items might not actually be removed.
+        self.session.checkpoint()
+
+        # Count the overflow items; should have dropped.
+        updated_overflow_items = self.get_tree_stat(ds, stat.dsrc.btree_overflow)
+        removed = initial_overflow_items - updated_overflow_items
+        self.assertGreater(removed, 0)
+
+        # Check how many on-disk pages we read in to do this. Should be > 0.
+        pages_read = self.get_stat(stat.conn.cc_pages_read)
+        self.assertGreater(pages_read, 0)
+
+if __name__ == '__main__':
+    wttest.run()
+#!/usr/bin/env python
```

Comment:
The content seems to be duplicated from here

Reply:
Wonder how that happened... fixed in bd5b940